tools/tdbg/adminCommands.go

```diff
@@ -0,0 +1,744 @@
+		allEvents.Events = append(allEvents.Events, historyBatch...)
+		encoder := codec.NewJSONPBEncoder()
+		data, err := encoder.EncodeHistoryEvents(historyBatch)
+		if err != nil {
+			return fmt.Errorf("EncodeHistoryEvents err: %s", err)
+		}
+		fmt.Println(string(data))
+	}
+	fmt.Printf("======== total batches %v, total blob len: %v ======\n", len(history), totalSize)
+
+	if outputFileName != "" {
+		encoder := codec.NewJSONPBEncoder()
+		data, err := encoder.EncodeHistoryEvents(allEvents.Events)
+		if err != nil {
+			return fmt.Errorf("Failed to serialize history data: %s", err)
+		}
+		if err := os.WriteFile(outputFileName, data, 0666); err != nil {
+			return fmt.Errorf("Failed to export history data file: %s", err)
+		}
+	}
+	return nil
+}
+
+// AdminDescribeWorkflow describe a new workflow execution for admin
+func AdminDescribeWorkflow(c *cli.Context) error {
+	resp, err := describeMutableState(c)
+	if err != nil {
+		return err
+	}
+
+	if resp != nil {
+		fmt.Println(color.Green(c, "Cache mutable state:"))
+		if resp.GetCacheMutableState() != nil {
+			prettyPrintJSONObject(resp.GetCacheMutableState())
+		}
+		fmt.Println(color.Green(c, "Database mutable state:"))
+		prettyPrintJSONObject(resp.GetDatabaseMutableState())
+
+		fmt.Println(color.Green(c, "Current branch token:"))
+		versionHistories := resp.GetDatabaseMutableState().GetExecutionInfo().GetVersionHistories()
+		// if VersionHistories is set, then all branch infos are stored in VersionHistories
+		currentVersionHistory, err := versionhistory.GetCurrentVersionHistory(versionHistories)
+		if err != nil {
+			fmt.Println(color.Red(c, "Unable to get current version history:"), err)
+		} else {
+			currentBranchToken := persistencespb.HistoryBranch{}
+			err := currentBranchToken.Unmarshal(currentVersionHistory.BranchToken)
+			if err != nil {
+				fmt.Println(color.Red(c, "Unable to unmarshal current branch token:"), err)
+			} else {
+				prettyPrintJSONObject(currentBranchToken)
+			}
+		}
+
+		fmt.Printf("History service address: %s\n", resp.GetHistoryAddr())
+		fmt.Printf("Shard Id: %s\n", resp.GetShardId())
+	}
+	return nil
+}
+
+func describeMutableState(c *cli.Context) (*adminservice.DescribeMutableStateResponse, error) {
+	adminClient := cFactory.AdminClient(c)
+
+	namespace, err := getRequiredGlobalOption(c, FlagNamespace)
+	if err != nil {
+		return nil, err
+	}
+	wid := c.String(FlagWorkflowID)
+	rid := c.String(FlagRunID)
+
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	resp, err := adminClient.DescribeMutableState(ctx, &adminservice.DescribeMutableStateRequest{
+		Namespace: namespace,
+		Execution: &commonpb.WorkflowExecution{
+			WorkflowId: wid,
+			RunId:      rid,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Get workflow mutableState failed: %s", err)
+	}
+	return resp, nil
+}
+
+// AdminDeleteWorkflow delete a workflow execution from Cassandra and visibility document from Elasticsearch.
+func AdminDeleteWorkflow(c *cli.Context) error {
```

Comment:
This need to use delete workflow API from admin service, we don't want direct db access here.

Reply:
I can make the change. Already have the changes on my local when testing the admin delete API.

Reply:
replaced DeleteWorkflow with `not implemented` for @yycptt to replace with the new implementation. cc @alexshtin